src/fu-engine.c

```diff
@@ -6366,6 +6372,7 @@ fu_engine_backend_device_removed_cb(FuBackend *backend, FuDevice *device, FuEngi
 				fu_device_get_name(device_tmp),
 				fu_device_get_id(device_tmp));
 			fu_device_list_remove(self->device_list, device_tmp);
+			fu_engine_emit_changed(self);
```

Comment:
I wonder if we should deprecate/drop `enable-json-state` now for fwupdtool.  This is effectively done by all updates this means from fwupd or fwupdtool.

Reply:
I do not mind deprecating that option; I wonder if people is using it.

Reply:
> I do not mind deprecating that option; I wonder if people is using it.

Yeah let's please do that as part of this PR.

Reply:
what does deprecating the option imply?

Adds a warning in a first step, or drop all the code off directly?

Reply:
> what does deprecating the option imply?
> 
> Adds a warning in a first step, or drop all the code off directly?

I think you can just drop all the code and just have it emit a message when that parameter is used can't you?  Since you're writing out that file by default now.

Reply:
Done.

@superm1, could you check I have not missed something please?